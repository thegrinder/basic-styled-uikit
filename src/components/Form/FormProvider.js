import { createProvider } from '../../helpers';

const FormProvider = createProvider('form', {});

export default FormProvider;
