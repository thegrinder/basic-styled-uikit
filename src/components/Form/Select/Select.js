import React from 'react';
import styled from 'styled-components';
import { bool } from 'prop-types';

import { getSharedInputStyles } from '../commonFormStyles';
import { rem } from '../../../theme/typography';

const propTypes = {
  /** invalid flag */
  invalid: bool,
};

const StyledSelect = styled.select(({ theme, invalid }) => ({
  textTransform: 'none',
  padding: `0 ${rem(theme, 10)}`,
  verticalAlign: 'middle',
  display: 'inline-block',
  height: rem(theme, 40),
  backgroundColor: '#fff',
  ...getSharedInputStyles(theme, invalid),
}));

StyledSelect.propTypes = propTypes;

const defaultProps = {
  invalid: false,
};

const Select = props => <StyledSelect {...props} />;

Select.propTypes = propTypes;
Select.defaultProps = defaultProps;

export default Select;
