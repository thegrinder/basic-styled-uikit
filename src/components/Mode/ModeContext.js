import { createContext } from 'react';

const ModeContext = createContext(undefined);

export default ModeContext;
