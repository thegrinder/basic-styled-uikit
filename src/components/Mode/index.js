export { default as ModeProvider } from './ModeProvider';
export { default as useMode } from './useMode';
