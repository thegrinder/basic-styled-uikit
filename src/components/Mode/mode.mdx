---
name: Mode
menu: Components
route: /components/mode
---

## **ModeProvider**



---



## **useMode**
