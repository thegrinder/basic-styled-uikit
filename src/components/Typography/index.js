export { default as TypographyProvider } from './TypographyProvider';
export { default as Text } from './Text/Text';
export * from './typographySelectors';
