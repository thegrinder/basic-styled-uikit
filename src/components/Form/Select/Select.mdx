---
name: Select
menu: Components
---

import { Playground, Props } from 'docz';
import Select from './Select';
import { InputWrapper } from '../../../../docz';



# Select

<Props of={Select} />


### Select

<Playground>
  <InputWrapper>
    <Select>
      <option>option one</option>
      <option>option two</option>
    </Select>
  </InputWrapper>
</Playground>


### Invalid Select

<Playground>
  <InputWrapper>
    <Select invalid={true}>
      <option>option one</option>
      <option>option two</option>
    </Select>
  </InputWrapper>
</Playground>


### Disabled Select

<Playground>
  <InputWrapper>
    <Select disabled={true}>
      <option>option one</option>
      <option>option two</option>
    </Select>
  </InputWrapper>
</Playground>
