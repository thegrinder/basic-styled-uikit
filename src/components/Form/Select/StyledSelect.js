import styled from 'styled-components';
import { bool } from 'prop-types';
import { commonInputStyles } from '../commonFormStyles';
import { rem } from '../../Typography';

const propTypes = {
  invalid: bool.isRequired,
};

const StyledSelect = styled.select`
  ${commonInputStyles}
  text-transform: none;
  padding: 0 ${rem(10)};
  vertical-align: middle;
  display: inline-block;
  height: ${rem(40)};
`;

StyledSelect.propTypes = propTypes;

export default StyledSelect;
