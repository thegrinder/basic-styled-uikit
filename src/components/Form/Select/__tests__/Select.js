import React from 'react';
import { render } from '@testing-library/react';
import { ModeProvider } from '../../../Mode';
import { TypographyProvider } from '../../../Typography';
import FormProvider from '../../FormProvider';
import Select from '../Select';

const testId = 'select';
const option = <option>option</option>;

const renderComponent = (props = {}) =>
  render(
    <ModeProvider>
      <TypographyProvider>
        <FormProvider>
          <Select data-testid={testId} {...props}>
            {option}
          </Select>
        </FormProvider>
      </TypographyProvider>
    </ModeProvider>
  );

describe('<Select />', () => {
  it('should render correctly with default props and its children', () => {
    const { queryByTestId, queryByText } = renderComponent();
    expect(queryByTestId(testId)).toBeTruthy();
    expect(queryByText('option')).toBeTruthy();
  });

  it('should render correctly with custom props', () => {
    const { queryByTestId } = renderComponent({
      invalid: true,
    });
    expect(queryByTestId(testId)).toBeTruthy();
  });

  it('should render <select> tag', () => {
    const { queryByTestId } = renderComponent();
    expect(queryByTestId(testId).tagName).toEqual('SELECT');
  });
});
