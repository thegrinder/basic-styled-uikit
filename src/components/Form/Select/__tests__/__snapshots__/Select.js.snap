// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render correctly with custom props 1`] = `
<select
  class="StyledSelect-sc-1r3g39b-0 htlzwZ"
>
  <option>
    option
  </option>
</select>
`;

exports[`<Select /> should render correctly with default props and its children 1`] = `
<select
  class="StyledSelect-sc-1r3g39b-0 kdPVPy"
>
  <option>
    option
  </option>
</select>
`;
